src/transformers/models/audio_spectrogram_transformer/feature_extraction_audio_spectrogram_transformer.py

```diff
@@ -230,7 +247,39 @@ def __call__(
         if self.do_normalize:
             padded_inputs["input_values"] = [self.normalize(feature) for feature in input_values]
 
+        if add_noise:
+            padded_inputs["input_values"] = [self.add_noise(feature) for feature in input_values]
+
         if return_tensors is not None:
             padded_inputs = padded_inputs.convert_to_tensors(return_tensors)
 
         return padded_inputs
+
+    # from https://github.com/YuanGongND/ast with minor changes
+    def mask(self, fbank, freqm, timem):
```

Comment:
Let's also rename this method to a more explicit name ! (i.e `apply_time_frequency_masks`) ?